### ISSUE TITLE
Implicit Translations

### ISSUE THREAD

**Issue by donaldvaneupen** [phrase/phraseapp-in-context-editor-ruby#17]
Good morning,

We're currently trying to implement the editor into our app, and we're running into an issue here. Due to the nature of our app, we often fall back to Rails' i18n own fallback mechanisms. I.e., when we use forms with Simple Form, we do not explicitly add keys in the simple_form nesting to render the labels, but rather, we expect i18n to fall back onto the Model translations (which Simple Form supports out of the box). 
So for example:

```
      <%= f.input :email %>
      <%= f.input :password %>
      <% if devise_mapping.rememberable? %>
          <%= f.input :remember_me, as: :boolean, wrapper: :horizontal_boolean %>
      <% end %>
```
Should've just fallen back to the Login's model translations, but it
gave us the following error message:
```
uncaught throw :exception (pointing to the `<%= f.input :email %>` line)
```
Following the stack trace, that led us to:
```
2016-11-28 10:51:26.222 [WRN] phrase: Trying to execute missing method #to_hash on key simple_form.hints.admin.new.email 
2016-11-28 10:51:26.491 [WRN] phrase: You tried to execute the missing method #to_hash on key simple_form.hints.admin.new.email which is not supported. Please make sure you treat your translations as strings only. 

2016-11-28 10:51:26.493 [WRN] phrase: Trying to execute missing method #to_hash on key simple_form.hints.admin.email 
2016-11-28 10:51:26.777 [WRN] phrase: You tried to execute the missing method #to_hash on key simple_form.hints.admin.email which is not supported. Please make sure you treat your translations as strings only. 
```

We've been trying to locate where and how the exception is thrown in Rails, Simple_Form as well as your source code, but weren't able to find it. Initially we thought it might have been due to us having:
```
config.action_view.raise_on_missing_translations = true
```
but we tested it both ways and the exceptions kept being thrown. Effectively, we do not have those keys that are mentioned due to the fallbacks we're using. When we added those keys manually, the exception wasn't thrown anymore. However, that is a very undesirable option, considering the size of the app and the amount of duplication we would have to add in all the locale. 

For now we will try to ignore (or add, if not too many) all the nestings that cause issues, but we would be really grateful for any hint that might solve this. Our only suspicion at this point is that there is not support for Simple_Form's use of that fallback mechanism (which we've been using from the start, to keep it as DRY as possible). We also use other gems such as Ransack, which have that fallback mechanism, and solving this would help us a great deal implementing the editor.

Thanks already in advance for your time!

**Comment by docstun** [comment]
Thanks for reporting this. We're happy to have another look at possible issues with simple_form. Could you tell us which version of Rails, simple_form and the phraseapp gem you're using?

**Comment by donaldvaneupen** [comment]
Good morning,
thanks for the quick reply. Here are the versions:

Rails 4.2.7.1
Ruby 2.3.1
SimpleForm 3.2.0
Phraseapp-in-context-editor-ruby 1.1.0

Is there any other info you would need?

**Comment by docstun** [comment]
Thanks for sharing! We'll take a look at this as soon as we can and I will make sure we'll notify you once we know more!

**Comment by ml** [comment]
@docstun I think ran into the same issue:

```
td.col-md-2
  = order_item_form.input :service_type_id, as: "hidden", value: order_item.service_type_id
td.col-md-2
  = order_item_form.input :amount, label: false do
    = order_item_form.select :amount, ...
```

I get `uncaught throw :exception` from the line number 4. 

My stack:
ruby '2.3.1'
rails (4.2.6)
simple_form (3.2.0)
phraseapp-in-context-editor-ruby (1.1.0)

I'll appreciate any update on this, thanks.

**Comment by docstun** [comment]
@ml thanks for reporting this. Could you maybe share the whole stacktrace? If you don't want to share it publicly, you can of course send it to support@phraseapp.com

**Comment by docstun** [comment]
@donaldvaneupen could you tell me if the issue is fixed in this branch? https://github.com/phrase/phraseapp-in-context-editor-ruby/tree/fix-simple-form-issue

**Comment by donaldvaneupen** [comment]
Sure will do once I get time to have a look at it!

**Comment by docstun** [comment]
Fixed in #22.

**Comment by docstun** [comment]
Published now via https://rubygems.org/gems/phraseapp-in-context-editor-ruby/versions/1.2.1
